public/app/features/alerting/unified/mocks/server/db.ts

```diff
@@ -59,7 +56,49 @@ const prometheusRuleGroupFactory = Factory.define<PromRuleGroupDTO>(({ sequence
   return group;
 });
 
-const dataSourceFactory = Factory.define<DataSourceInstanceSettings>(({ sequence, params, afterBuild }) => {
+const rulerAlertingRuleFactory = Factory.define<RulerAlertingRuleDTO>(({ sequence }) => ({
+  alert: `test-rule-${sequence}`,
+  expr: 'up = 1',
+  labels: { severity: 'warning' },
+  annotations: { summary: 'test alert' },
+}));
+
+const rulerRecordingRuleFactory = Factory.define<RulerRecordingRuleDTO>(({ sequence }) => ({
+  record: `ruler-recording-rule-${sequence}`,
+  expr: 'vector(0)',
+  labels: {},
+}));
+
+const rulerGroupFactory = Factory.define<RulerRuleGroupDTO<RulerCloudRuleDTO>, { addToNamespace: string }>(
+  ({ sequence, transientParams, afterBuild }) => {
+    afterBuild((group) => {
+      if (transientParams.addToNamespace) {
+        if (!namespaces[transientParams.addToNamespace]) {
+          namespaces[transientParams.addToNamespace] = [];
+        }
+        namespaces[transientParams.addToNamespace].push(group);
+      }
+    });
```

Comment:
I'm not sure if we should do it this way. I think I'd prefer a more explicit approach, where I need to set the namespace in the test, wdyt?
I'm not sure how to balance relying on predefined test data and defining custom ones which are sometimes necessary to test properly

Reply:
I think it's ok, otherwise we might build an inconsistent data?